llvm/test/CodeGen/AIE/aie2/schedule/swp/swp-regsequence.mir

```diff
@@ -0,0 +1,216 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+
+# RUN: llc --mtriple=aie2 --run-pass=pipeliner %s -o - \
+# RUN: --aie-loop-min-tripcount=2 --aie-pipeliner-max-stagecount=2 --aie-postpipeliner-limit=1 | FileCheck %s
+
+--- |
+
+  define void @vmul_256(ptr noalias %out, ptr noalias readonly %in1, ptr noalias readonly %in2) {
+  entry:
+    ret void
+  }
+  define void @vmul_512(ptr noalias %out, ptr noalias readonly %in1, ptr noalias readonly %in2) {
+  entry:
+    ret void
+  }
+...
+
+# Here we verify in which stage REG_SEQUENCE is placed after SW pipelining.
+# This is mostly driven by the PropagateIncomingLatencies DAGMutator.
+
+# When we only load 256 bit into a 512 bit vector, and the rest comes from an
+# external source to fill the hi bits, we would like the REG_SEQUENCE to be
+# placed close to consumer to avoid making the whole 512 bit register
+# loop-carried and facilitate LICM for the constant part.
+# When the whole 512 bit register is defined in the loop, we would like to keep
+# the REG_SEQUENCE close to its sources to facilitate register coalescing and
+# "tie" the sources together.
+
+# We expect the REG_SEQUENCE for the load of %ir.in1 to be in the second stage, close to
+# its VMUL consumer.
```

Comment:
Perhaps say explicitly you don't want the VMUL in the steady state to read a PHI node, but rather the REG_SEQUENCE